src/bundle/ArgumentResolver/SuggestionQueryArgumentResolver.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+/**
+ * @copyright Copyright (C) Ibexa AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Bundle\Search\ArgumentResolver;
+
+use Ibexa\Contracts\Core\SiteAccess\ConfigResolverInterface;
+use Ibexa\Search\Model\SuggestionQuery;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+use Symfony\Component\HttpKernel\ControllerMetadata\ArgumentMetadata;
+use Symfony\Component\HttpKernel\Exception\BadRequestHttpException;
+
+final class SuggestionQueryArgumentResolver implements ArgumentValueResolverInterface
+{
+    private int $defaultLimit;
+
+    public function __construct(ConfigResolverInterface $configResolver)
+    {
+        $this->defaultLimit = $configResolver->getParameter('search.suggestion.result_limit');
```

Comment:
I am almost sure that this not works properly when siteaccess is changed during runtime, and it should be not set in constructor but resolved when needed.

Reply:
Fixed